conda_forge_tick/update_upstream_versions.py

```diff
@@ -417,38 +419,43 @@ def get_url(self, meta_yaml) -> Optional[str]:
             found = False
             for next_ver in next_version(current_ver):
                 logger.debug("trying version: %s", next_ver)
+
                 new_content = content.replace(orig_ver, next_ver)
-                meta = parse_meta_yaml(new_content)
-                url = None
-                for u in urls_from_meta(meta):
-                    if u not in orig_urls:
-                        url = u
-                        break
-                if url is None:
+                new_meta = parse_meta_yaml(new_content)
+                new_urls = urls_from_meta(new_meta)
+                if len(new_urls) == 0:
+                    logger.debug("No URL in meta.yaml")
                     meta_yaml["bad"] = "Upstream: no url in yaml"
                     return None
-                if (
-                    str(meta["package"]["version"]) != next_ver
-                    or meta_yaml["url"] == url
-                ):
-                    continue
-
-                _exists, _url_to_use = url_exists_swap_exts(url)
-                if not _exists:
-                    logger.debug("version %s does not exist", next_ver)
-                    continue
-                else:
-                    url = _url_to_use
 
-                found = True
-                count = count + 1
-                current_ver = next_ver
-                new_sha256 = get_sha256(url)
-                if new_sha256 == current_sha256 or new_sha256 in new_content:
-                    return None
-                current_sha256 = new_sha256
-                logger.debug("version %s is ok", current_ver)
-                break
+                url_to_use = None
+                for url in urls_from_meta(new_meta):
+                    # this URL looks bad if these things happen
+                    if (
+                        str(new_meta["package"]["version"]) != next_ver
+                        or meta_yaml["url"] == url
```

Comment:
Is  `meta_yaml["url"]` a list ?

Reply:
This is the top-level key in the attributes, not the actual meta_yaml. This code prexists me and so I didn't want to change it.